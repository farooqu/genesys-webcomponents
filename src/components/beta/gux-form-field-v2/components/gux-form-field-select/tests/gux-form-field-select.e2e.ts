import { E2EPage, newE2EPage } from '@stencil/core/testing';
import {
  newSparkE2EPage,
  a11yCheck
} from '../../../../../../../tests/e2eTestUtils';

const axeExclusions = [];

async function newNonrandomE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();

  return page;
}

describe('gux-form-field-select-beta', () => {
  describe('#render', () => {
    describe('label-position', () => {
      [
        '',
        'label-position="above"',
        'label-position="beside"',
        'label-position="screenreader"'
      ].forEach((componentAttribute, index) => {
        const html = `
          <gux-form-field-select-beta ${componentAttribute}>
            <select slot="input" name="lp-1">
              <option value="option1">Option 1</option>
              <option value="option2">Option 2</option>
              <option value="option3">Option 3</option>
            </select>
            <label slot="label">Default</label>
          </gux-form-field-select-beta>
        `;

        it(`should render component as expected (${index + 1})`, async () => {
          const page = await newNonrandomE2EPage({ html });
          const element = await page.find('gux-form-field-select-beta');
          const elementShadowDom = await element.find(
            'pierce/.gux-form-field-container'
          );

          expect(element.outerHTML).toMatchSnapshot();
          expect(elementShadowDom).toMatchSnapshot();
        });

        it(`should be accessible (${index + 1})`, async () => {
          const page = await newSparkE2EPage({ html });

          await a11yCheck(page, axeExclusions);
        });
      });
    });

    describe('input attributes', () => {
      ['', 'disabled', 'required'].forEach((inputAttribute, index) => {
        const html = `
          <gux-form-field-select-beta>
            <select slot="input" name="lp-1" ${inputAttribute}>
              <option value="option1">Option 1</option>
              <option value="option2">Option 2</option>
              <option value="option3">Option 3</option>
            </select>
            <label slot="label">Default</label>
          </gux-form-field-select-beta>
        `;

        it(`should render component as expected (${index + 1})`, async () => {
          const page = await newNonrandomE2EPage({ html });
          const element = await page.find('gux-form-field-select-beta');
          const elementShadowDom = await element.find(
            'pierce/.gux-form-field-container'
          );

          expect(element.outerHTML).toMatchSnapshot();
          expect(elementShadowDom).toMatchSnapshot();
        });

        it(`should be accessible (${index + 1})`, async () => {
          const page = await newSparkE2EPage({ html });

          await a11yCheck(page, axeExclusions);
        });
      });
    });
  });
});
