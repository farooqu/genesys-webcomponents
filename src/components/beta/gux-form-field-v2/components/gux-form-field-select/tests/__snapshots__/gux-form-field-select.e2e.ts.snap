// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field-select-beta #render input attributes should render component as expected (1) 1`] = `"<gux-form-field-select-beta class=\\"hydrated\\"> <select slot=\\"input\\" name=\\"lp-1\\" id=\\"gux-form-field-input-i\\"> <option value=\\"option1\\">Option 1</option> <option value=\\"option2\\">Option 2</option> <option value=\\"option3\\">Option 3</option> </select> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> </gux-form-field-select-beta>"`;

exports[`gux-form-field-select-beta #render input attributes should render component as expected (1) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <slot name="input"></slot>
        <gux-icon class="hydrated" icon-name="ic-dropdown-arrow"></gux-icon>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-select-beta #render input attributes should render component as expected (2) 1`] = `"<gux-form-field-select-beta class=\\"hydrated\\"> <select slot=\\"input\\" name=\\"lp-1\\" disabled=\\"\\" id=\\"gux-form-field-input-i\\"> <option value=\\"option1\\">Option 1</option> <option value=\\"option2\\">Option 2</option> <option value=\\"option3\\">Option 3</option> </select> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> </gux-form-field-select-beta>"`;

exports[`gux-form-field-select-beta #render input attributes should render component as expected (2) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-disabled gux-input-container">
        <slot name="input"></slot>
        <gux-icon class="hydrated" icon-name="ic-dropdown-arrow"></gux-icon>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-select-beta #render input attributes should render component as expected (3) 1`] = `"<gux-form-field-select-beta class=\\"hydrated\\"> <select slot=\\"input\\" name=\\"lp-1\\" required=\\"\\" id=\\"gux-form-field-input-i\\"> <option value=\\"option1\\">Option 1</option> <option value=\\"option2\\">Option 2</option> <option value=\\"option3\\">Option 3</option> </select> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> </gux-form-field-select-beta>"`;

exports[`gux-form-field-select-beta #render input attributes should render component as expected (3) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label gux-required">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <slot name="input"></slot>
        <gux-icon class="hydrated" icon-name="ic-dropdown-arrow"></gux-icon>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-select-beta #render label-position should render component as expected (1) 1`] = `"<gux-form-field-select-beta class=\\"hydrated\\"> <select slot=\\"input\\" name=\\"lp-1\\" id=\\"gux-form-field-input-i\\"> <option value=\\"option1\\">Option 1</option> <option value=\\"option2\\">Option 2</option> <option value=\\"option3\\">Option 3</option> </select> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> </gux-form-field-select-beta>"`;

exports[`gux-form-field-select-beta #render label-position should render component as expected (1) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <slot name="input"></slot>
        <gux-icon class="hydrated" icon-name="ic-dropdown-arrow"></gux-icon>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-select-beta #render label-position should render component as expected (2) 1`] = `"<gux-form-field-select-beta label-position=\\"above\\" class=\\"hydrated\\"> <select slot=\\"input\\" name=\\"lp-1\\" id=\\"gux-form-field-input-i\\"> <option value=\\"option1\\">Option 1</option> <option value=\\"option2\\">Option 2</option> <option value=\\"option3\\">Option 3</option> </select> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> </gux-form-field-select-beta>"`;

exports[`gux-form-field-select-beta #render label-position should render component as expected (2) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <slot name="input"></slot>
        <gux-icon class="hydrated" icon-name="ic-dropdown-arrow"></gux-icon>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-select-beta #render label-position should render component as expected (3) 1`] = `"<gux-form-field-select-beta label-position=\\"beside\\" class=\\"hydrated\\"> <select slot=\\"input\\" name=\\"lp-1\\" id=\\"gux-form-field-input-i\\"> <option value=\\"option1\\">Option 1</option> <option value=\\"option2\\">Option 2</option> <option value=\\"option3\\">Option 3</option> </select> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> </gux-form-field-select-beta>"`;

exports[`gux-form-field-select-beta #render label-position should render component as expected (3) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <slot name="input"></slot>
        <gux-icon class="hydrated" icon-name="ic-dropdown-arrow"></gux-icon>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-select-beta #render label-position should render component as expected (4) 1`] = `"<gux-form-field-select-beta label-position=\\"screenreader\\" class=\\"hydrated\\"> <select slot=\\"input\\" name=\\"lp-1\\" id=\\"gux-form-field-input-i\\"> <option value=\\"option1\\">Option 1</option> <option value=\\"option2\\">Option 2</option> <option value=\\"option3\\">Option 3</option> </select> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> </gux-form-field-select-beta>"`;

exports[`gux-form-field-select-beta #render label-position should render component as expected (4) 2`] = `
<div class="gux-form-field-container gux-screenreader">
  <div class="gux-form-field-label gux-screenreader">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <slot name="input"></slot>
        <gux-icon class="hydrated" icon-name="ic-dropdown-arrow"></gux-icon>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;
